### DISCUSSION TITLE
What data to add next?

### DISCUSSION THREAD

**Discussion by mledoze** [mledoze/countries#501] [Ideas]
I would like to discuss here the data that should be added to this repository. 

A similar project like 0xJS [1] contains a lot more data such as the land area or the latitude/longitude coordinates of each country. 

Is it interesting/useful to have this kind of data too? 

Data that can be added:
- land (land mass in square kilometers [3])
- latitude (latitude coordinate of the capital [2])
- longitude (longitude coordinate of the capital [2])
- east (longitude of the country's eastern boundary [3])
- north (latitude of the country's northern boundary [3])
- south (latitude of the country's southern boundary [3])

What would you like to be added? 

Please let me know in the comments. 

[1] http://oxjs.org/#doc/Ox.COUNTRIES
[2] source: http://opengeocode.org/
[3] source: https://oxjs.org/#doc/Ox.COUNTRIES

---

**From the comments**
- add the type of the country (country, sovereign state, public body, territory, etc.)
- add the land borders (**done**, see https://github.com/mledoze/countries/tree/v1.3)
- add regions, provinces and cities

**Comment by scento** [comment]
It might be useful to provide the country name in the native language of the country itself (e.g. `{"name": "Germany", "name_native": "Deutschland"}`...

**Comment by scento** [comment]
The CLDR database of the unicode project contains Country-To-Language data, including the percent of speakers: http://www.unicode.org/cldr/charts/latest/supplemental/territory_language_information.html

**Comment by mledoze** [comment]
> It might be useful to provide the country name in the native language of the country itself

The native name of Germany is already in 'alt-spellings'. I recognize that the name 'alt-spellings' isn't good since it contains alternative spellings and the native name of the country. So there are two solutions here:
- either we change 'alt-spellings' to 'alt-names' and keep the native name here
- or we keep 'alt-spellings' just for alternative spellings and create 'name-native' as you suggested.

Initially, I created this dataset with a country selector in mind [1] but it would make more sense to be able to get the native names separately. So I would choose the second option. 

But the second option raises the question of how to write the native name of the country. German uses latin characters so it's easy to know that it's Germany, but what about Armenia for example which is written Հայաստան in armenian [2]? For some people it might be difficult to know that it's Armenia. 

What do you think?

I know that alternative spellings and native names are missing for many countries, I'm currently working on adding them. Also, I'll add the native/official language(s) of each country. 

[1] https://github.com/JamieAppleseed/selectToAutocomplete
[2] http://en.wikipedia.org/wiki/Armenia

**Comment by scento** [comment]
Not all people speak English, so they might be confused while selecting their locale. It might be useful if it is possible to see the English and native version of the country name parallel in the selector.

I would recommend to provide both versions for different individual usecases.

**Comment by mledoze** [comment]
Right, it's valid for non english speakers.

**If you want**, feel free to start working on adding the native names as I'll be off for a few days.

**Comment by stephenpaulger** [comment]
I think it would be great to have a way to make Countries Hierarchical and have meta data describing whether they are countries or sovereign states.

For the UK currently it says "alt-spellings":"GB,Great Britain,England,UK,Wales,Scotland,Northern Ireland".

The full name of the UK is "The United Kingdom of Great Britain and Northern Ireland". It is not a country, it is a sovereign state.

Great Britain also isn't a country, it's an island.

There are three countries in Great Britain: England, Scotland and Wales.

So the types I think needed are: Country, State, Sovereign State and potentially Nation and Union as well.

Then it would be good to have a way to specify that England is within the UK and if you also have unions that it is within the EU.

Another nice feature would be to list what land borders a country has. So you could specify that England borders Scotland and Wales for example.

**Comment by fayderflorez** [comment]
From https://github.com/ProGNOMmers

It would be wonderful if it would be possible to retrieve regions, provinces and cities.

Something like:

``` javascript
// Regions of country
// /rest/alpha2/it/regions ->
{ regions:  [ "Abruzzi e Molise",
              "Basilicata",
              "Calabria",
              "Campania",
              "Emilia-Romagna",
              "Friuli-Venezia Giulia",
              "Lazio",
              "Liguria",
              "Lombardia",
              "Marche",
              "Piemonte",
              "Puglia",
              "Sardegna",
              "Sicilia",
              "Toscana",
              "Trentino-Alto Adige",
              "Umbria",
              "Valle d'Aosta",
              "Veneto" ] }

// Provinces of region
// /rest/alpha2/it/regions/Veneto/provinces ->
{ provinces: [ "Verona", "Venezia", ... ] }

// Cities of province
// /rest/alpha2/it/regions/Veneto/provinces/Venezia/cities ->
{ cities: [ { name: "Venezia", zip_codes: [ "30121", ... , "30176" ] }, 
            { name: "Chioggia", zip_codes: [ "30015" ] },
            { name: "San Donà di Piave", zip_codes: [ "30027" ] }, 
            ... ] }

// Cities of country by name
// /rest/alpha2/it/regions/Veneto/provinces/Venezia/cities ->
{ cities: [ { name: "Venezia", zip_codes: [ "30121", ... , "30176" ] }, 
            { name: "Chioggia", zip_codes: [ "30015" ] },
            { name: "San Donà di Piave", zip_codes: [ "30027" ] }, 
            ... ] }
```

Cities could have metadata like f.i. zip codes, which are very useful.

It is a huge work because recording and maintaining the whole list of regions, provinces and cities for every world country is hard, but it is a good target to be accomplished by an open source project.

**Comment by mledoze** [comment]
@stephenpaulger 

> I think it would be great to have a way to make Countries Hierarchical and have meta data describing whether they are countries or sovereign states.

I agree, I'll add this to the todo. I know that many entries in the dataset are not actual contries. I wanted to provide simple and factual data about world countries but I understand that more accuracy is needed.

**Comment by mledoze** [comment]
@fayder 

> It would be wonderful if it would be possible to retrieve regions, provinces and cities.

Yes it is a huge work. First I want to continue to add more data at the country level (native and official names, official language, etc.) and add the master file as soon as possible (#12) to ease the contributions.

Thank you for your help/feedback, I appreciate it!

**Comment by mledoze** [comment]
> For the UK currently it says "alt-spellings":"GB,Great Britain,England,UK,Wales,Scotland,Northern Ireland".

@stephenpaulger in bd22b4a97f30ead3ae55f68d2c3e9b86ba784ba7 I have removed most of the names in `altSpellings`, now it's just `GB,UK,Great Britain`.

**Comment by mledoze** [comment]
We can also add time zone data from http://timezonedb.com/download.

**Comment by shanti2530** [comment]
It would be really nice if there would be also a list of states per country such as the United States states. http://en.wikipedia.org/wiki/List_of_states_and_territories_of_the_United_States

**Comment by mledoze** [comment]
@shanti2530 yes, this has been suggested https://github.com/mledoze/countries/issues/6#issuecomment-27620009 but it has not been done yet because the work is pretty huge. 
Do you know a source where we can find the states for every country?

**Comment by shanti2530** [comment]
@mledoze don't know if this is what you were looking for http://vikku.info/programming/geodata/geonames-get-country-state-city-hierarchy.htm

**Comment by mledoze** [comment]
@shanti2530 this seems very good, thank you. I'll create an issue for this. Would you like to work on this?

**Comment by gerbenjacobs** [comment]
GeoJSON outlines of the countries: https://github.com/datasets/geo-boundaries-world-110m

**Comment by mledoze** [comment]
@gerbenjacobs yes good idea, I'll add this to the to-do

**Comment by oriolfg** [comment]
I agree for the gerbenjacobs idea of GeoJSON outlines of the countries

**Comment by matiassingers** [comment]
@mledoze don't know if it's in the scope of this project, but I would love to see financial information like GDP, GDP per capita, GNI etc. - problem with this is of course that these numbers would change every year.

**Comment by mledoze** [comment]
@matiassingers no it's not really in the scope of this project. I prefer to stick with static data that do not change. The dataset currently contains population data which are not in the scope and I would like to remove it in the near future.

Although it does not currently contains GDP data, you should check this project https://github.com/tinata/tinatapi which contains other financial data.

**Comment by mledoze** [comment]
@dalu the postal prefixes is a good idea!

**Comment by mledoze** [comment]
@dalu you are saying that postal services want the native country name instead of the country postal prefix?

**Comment by mledoze** [comment]
I would like to inform you that I am about to remove population data because they require frequent updates to stay relevant. 

I recently added [CONTRIBUTING](https://github.com/mledoze/countries/raw/master/CONTRIBUTING.md) explaining the contributions rules of this project. Population data do not follow these instructions.

**Comment by fayderflorez** [comment]
acknowledged

**Comment by tdegrunt** [comment]
How about the address format, from the page mentioned above: http://en.wikipedia.org/wiki/Address_(geography) 

This may be fairly difficult to do as it requires some pseudo templating language, so say for US: 
"addressFormat": "{{name}}\n{{houseNumber}} {{street}}\n{{locality}}\n{{city}}\n{{postalCode}}"

And would need agreement on the labels used...

**Comment by wires** [comment]
Hi, nice project! Thanks.

Something that would be useful to me is to know if a country is in the European Union.
(https://en.wikipedia.org/wiki/Member_state_of_the_European_Union)

This information is needed when you are a company in the EU dealing with international customers. If you charge VAT or not depends on whether your customer is in the EU or not.

If you are interested in including this information I could setup a pull-request

**Comment by mledoze** [comment]
@tdegrunt yes it is indeed a difficult task to do, but @hexorx managed to do it in his `countries` repository: https://github.com/hexorx/countries/blob/master/lib/data/countries.yaml

**Comment by mledoze** [comment]
@0x01 yes I'm interested in including this information. Could you please add it as extra data in the `data` folder using `[cca3].json` file names?

Thank you!

**Comment by wires** [comment]
@mledoze: I can do that (in `data` folder), but I think it makes more sense to put it into the main file. There is very little data added, basically a boolean whether or not it's an EU member state (and I'll leave out the field if it's false)

**Comment by mledoze** [comment]
@0x01 you are right that this represent little data but it would be useful for only 10% of the countries (26 member state of the EU out of 251 "countries" in the dataset).

Moreover, the EU is categorized as a supranational union and it exists many other unions in the world (see [1]), so as not to add many booleans in the main file, I prefer to add this data in separate files.

[1] http://en.wikipedia.org/wiki/Political_union#Supranational_and_continental_unions

**Comment by wires** [comment]
Hm, true. I redraw my offer for a pull request, as this EU membership is much more subtle indeed. Should I ever need to sort this out properly I'll come back with a pull request, but for now a simple list of country names is enough to get some rough indication. Which is good enough for my purposes. For example, this code in node does the trick

``` js
var rawCountries = require('countries.json');
var EU = [
    "Austria", "Belgium", "Bulgaria", "Croatia", "Cyprus",
    "Czech Republic", "Denmark", "Estonia", "Finland", "France", "Germany",
    "Greece", "Hungary", "Ireland", "Italy", "Latvia", "Lithuania",
    "Luxembourg", "Malta", "Netherlands", "Poland", "Portugal",
    "Romania", "Slovakia", "Slovenia", "Spain", "Sweden", "United Kingdom"
];

var countries = rawCountries.map(function(country){
    return { // for example
        code: country.cca3,
        name: country.name
        eu_member: _(EU).contains(country.name)
    }
});
```

**Comment by pelegm** [comment]
I'd add coastline length (CIA factbook field 2060).

**Comment by yackermann** [comment]
How about population??

**Comment by mledoze** [comment]
@herrniemand population data was already added to this dataset (https://github.com/mledoze/countries/commit/81fa9f68215d92fba2a850c272d019f539cf30ad) but later removed (see https://github.com/mledoze/countries/issues/6#issuecomment-42322804).

**Comment by yackermann** [comment]
I've seen you were discussing official names, and I got an idea: 

``` JSON
{
        "name": {
            "common": "Afghanistan",
            "native": "\u0627\u0641\u063a\u0627\u0646\u0633\u062a\u0627\u0646",
            "official": "Islamic Republic of Afghanistan"
        }
}
```

otherwise there to many <code>"name%insert type here%"</code>

**Comment by mledoze** [comment]
@herrniemand this is a very good idea. I also want to add the official name in its native language, so we could have something like this: 

``` JSON
{
        "name": {
            "common": "Afghanistan",
            "official": "Islamic Republic of Afghanistan",
            "native": {
                "common" : "\u0627\u0641\u063a\u0627\u0646\u0633\u062a\u0627\u0646",
                "official": "\u062f \u0627\u0641\u063a\u0627\u0646\u0633\u062a\u0627\u0646 \u0627\u0633\u0644\u0627\u0645\u064a \u062c\u0645\u0647\u0648\u0631\u06cc\u062a"
            }
        }
}
```

What do you think?

**Comment by yackermann** [comment]
@mledoze Yea. Awesome.

**Comment by yackermann** [comment]
@mledoze how about translations? I was thinking something like:

``` json
{
        "name": {
            "common": "Afghanistan",
            "official": "Islamic Republic of Afghanistan",
            "native": {
                "common" : "\u0627\u0641\u063a\u0627\u0646\u0633\u062a\u0627\u0646",
                "official": "\u062f \u0627\u0641\u063a\u0627\u0646\u0633\u062a\u0627\u0646 \u0627\u0633\u0644\u0627\u0645\u064a \u062c\u0645\u0647\u0648\u0631\u06cc\u062a"
            },
            "translations":{
                "ru":...,
                "de":...
            }
        }
}
```

or should we keep as it is?

**Comment by mledoze** [comment]
@herrniemand I prefer to keep the translations as it is for now.

**Comment by yackermann** [comment]
@mledoze ok.

**Comment by yackermann** [comment]
@mledoze. I've just stuck on a problem with native names. Some countries like Afghanistan and Åland Islands have more then one official language, so what do we define as `native`?

**Comment by mledoze** [comment]
@herrniemand for countries with more than one official language, you should use the language that is listed first in the `language` property. So for Afghanistan and Åland Islands, it is Pashto and Swedish respectively.

**Comment by yackermann** [comment]
@mledoze ok. Thanks.

**Comment by yackermann** [comment]
@mledoze Question about translations. What is the list of the languages we want country names would be translated? I suggest UN(7) official + first 15 from most speaking: https://en.wikipedia.org/wiki/List_of_languages_by_number_of_native_speakers

**Comment by yackermann** [comment]
@mledoze Another thing. Since we are changed structure of the language block, wouldn't it be correct to change translations to the same style:

``` json
"translations":{
    "de": {
        "common":"Russland",
        "official":"Russische Föderation"
    }...
}
```

?

**Comment by CyntaxDotDev** [comment]
Not  exactly contributing to this issue, the feature list or the roadmap, but I simply wanted to express my deepest thanks for putting this awesome list up on the web. I've been searching for it for over 2 years, and a simple though epicly effective Google search turned this page up.

Thank you for your amazing for and this amazing list of countries!

**Comment by mledoze** [comment]
@ReSpawN you're very welcome, thank you for your comment, I really appreciate it! This work would not be as it is now without the help of all the [contributors](https://github.com/mledoze/countries/graphs/contributors).

If you do something with this dataset, don't hesitate to add it the the [showcase list](https://github.com/mledoze/countries#showcase) in the readme.

**Comment by romsson** [comment]
A variable saying if the country is a landlocked country or not 
http://en.wikipedia.org/wiki/Landlocked_country

**Comment by pelegm** [comment]
+1 for @romsson's idea.

**Comment by yackermann** [comment]
+1 @romsson's idea *)

**Comment by mledoze** [comment]
@romsson that's ok for me! Would you like to do this?

**Comment by romsson** [comment]
I'll do it! Happy to contribute this this super useful project @mledoze

**Comment by mledoze** [comment]
For future contributions regarding country name translations, I found this CSV [1] which might help.

[1] http://jonahellison.com/21640-translated-country-names-unicode-csv

**Comment by yackermann** [comment]
@mledoze I think we need to update translations objects according to the "name" object.

``` json
"translations":{
    "de": {
        "common":"Russland",
        "official":"Russische Föderation"
    }...
}
```

**Comment by mledoze** [comment]
@herrniemand I'm ok with this change, although I'm reserved about the increased size of `countries.json`.

**Comment by Sjord** [comment]
I would like to know whether a country is in the [European Union](http://europa.eu/about-eu/countries/index_en.htm). Is that something suitable for this project to include for every country?

**Comment by mledoze** [comment]
@Sjord 

> I would like to know whether a country is in the European Union. Is that something suitable for this project to include for every country?

This has been asked here before (https://github.com/mledoze/countries/issues/6#issuecomment-46950625) and answered (https://github.com/mledoze/countries/issues/6#issuecomment-47659566). Unfortunately, this is not suitable to include in this project.

Though, thank you for asking!

**Comment by yackermann** [comment]
@mledoze I kept translation in one line to keep it smaller.

**Comment by mledoze** [comment]
> I kept translation in one line to keep it smaller.

@herrniemand I saw it, very good.

**Comment by mledoze** [comment]
Yay :smile:  I just got this short URL http://git.io/countries for this project!

**Comment by petert82** [comment]
Sweet! :dancers:

**Comment by yackermann** [comment]
@mledoze YAY!!!!!!!!!XD

**Comment by shehi** [comment]
I strongly would advise the addition of provinces and major cities. There already exist few packages in Github, with limited range of countries of course, such as:

https://github.com/astockwell/countries-and-provinces-states-regions
https://github.com/substack/provinces

If licenses are compatible, they could be included within this package with due credits wherever applicable.

**Comment by yackermann** [comment]
@shehi #40 is the discussion about that. And @fayder proposed [here(comment)](https://github.com/mledoze/countries/issues/6#issuecomment-27620009) the structure.

**Comment by yackermann** [comment]
@mledoze I just posted it on echojs... and it gone... XD
![tumblr_inline_nglo2t4jud1raprkq](https://cloud.githubusercontent.com/assets/1636116/6037427/67ec2f5e-acb8-11e4-9dd8-d74e431c0493.gif)

**Comment by mledoze** [comment]
@herrniemand yay thank you:) http://www.echojs.com/news/13325

**Comment by yackermann** [comment]
@mledoze XD Thank you dude! Bloody brilliant project! XD

**Comment by mledoze** [comment]
Yesterday was a good day, look at the stats: 
![capture](https://cloud.githubusercontent.com/assets/576509/6038000/991d224a-ac59-11e4-95d7-c4560a08c029.PNG)

**Comment by yackermann** [comment]
@mledoze its maintainers only view *( Send screenshot.

**Comment by yackermann** [comment]
@mledoze Nice! *)

**Comment by qweek** [comment]
Add sea borders

**Comment by cumanacr** [comment]
Hi, super useful project! 
Thanks for maintaining it.

I'm aware that population data were added it, then, were removed on:

> @herrniemand population data was already added to this dataset (81fa9f6) but later removed (see #6 (comment)).

But, What about include population historical data only?

Like this:

``` javascript
        "name": {
            "common": "Costa Rica",
            "official": "Republic of Costa Rica",
            "native": {
                "spa": {
                    "official": "Rep\u00fablica de Costa Rica",
                    "common": "Costa Rica"
                }
            }
        },
.
.
.
        "population": {
            "2013": { "unit": "million", "amount":"4.872" },
            "2012": { "unit": "million", "amount":"4.805" },
            "2011": { "unit": "million", "amount":"4.738" },
            "2010": { "unit": "million", "amount":"4.67" },
        }
 }
```

And we can get another interesting historical info for each country from [google](https://www.google.com/search?client=ubuntu&hs=ESQ&channel=fs&q=costa+rica+population&stick=H4sIAAAAAAAAAGOovnz8BQMDw2MOHiUO_Vx9A8MCi2ItrexkK_2c_OTEksz8PP3iEiBdXJKZnJgTX5SaDhSyKsgvKM0By3YxmnHBNQqRo9G0sMKQLI1G2RbZpGjs16o74nctpOFW9fnrLo2rxL55tOwGAKh4gTsAAQAA&sa=X&ei=TwPtVNrfB4SZNoGwg_AJ&ved=0CCEQth8wAA&biw=1920&bih=851) like:
- Life expectancy
- GDP per capita
- Fertility rate

Same pattern

``` javascript
        "lifeExpectancy": {
            "2012": { "unit": "year", "amount":"79.7" },
            "2011": { "unit": "year", "amount":"79.49" },
            "2010": { "unit": "year", "amount":"79.28" },
        }
```

What do you think?
I would be glad to help with this

Regards

**Comment by mledoze** [comment]
Hello @cumanacr, thank you for your comment but unfortunately this project does not contains historical data. One good candidate for this kind of data would be the [tinata](https://github.com/tinata/tinata) project, you could ask there.

**Comment by cumanacr** [comment]
@mledoze thanks, but no thanks, I like this project :smiley: 

Maybe a separate file like #40 proposes for the country regions but on historical data, eg:
- cri.historicaldata.json
- cri.hdata.json
- cri.hd.json

This way, countries.json keeps small.

maybe something like that, is better?

Regards

**Comment by JacquesMarais** [comment]
Multiple Capitals.

**Comment by Purus** [comment]
We have just the list of languages. How about adding official languages? For example, India has too many languages and there are around 15+ official languges out of all.

**Comment by mledoze** [comment]
@Purus the `languages` property already contains the list of official languages for each country (see [README](https://github.com/mledoze/countries/blob/a0f1c99f7c7217a06b14871463dbc6e75dbdd603/README.md)). I know that there are some countries with many recognised regional languages, I would prefer no to take them into account as it would make the dataset too big.

**Comment by lifenautjoe** [comment]
How about adding the currencies names?

e.g. 

``` js
    "currencies" : {
      "eur" : "Euro",
      "mxn" : "Peso"
    }
```

**Comment by petert82** [comment]
@thefabulousdev For me, that's something that's better solved by cross-reference to a dedicated data-set. Something like [this](https://github.com/mhs/world-currencies/blob/master/currencies.json) for example.

**Comment by yackermann** [comment]
@petert82 agree

**Comment by BaseMax** [comment]
can you add language direction? ltr and rtl ?

**Comment by BaseMax** [comment]
can you add language direction? ltr and rtl ?

**Comment by pjona** [comment]
Hey,

would you be interested to add Polish translation?
I could prepare commit with data from [this list](https://gist.github.com/lukaszfiszer/7185871).

**Comment by yackermann** [comment]
We can. But not using that list. It is mixed up. There no distinction
between official name and common name, such as
- Rosya(Russia) - common
- Republika Południowej Afryki(Republic of South Africa) - official

2015-08-21 12:26 GMT+12:00 Marek Skiba notifications@github.com:

**Comment by mledoze** [comment]
@pjona 

> would you be interested to add Polish translation?

That could be a good addition, but @herrniemand is right we will need both common and official names.

**Comment by TuningGuide** [comment]
I would like to have the ISO639-1 code (de for Austria) in the list.

**Comment by jhourlad** [comment]
Lovely. I would have used it if it has timezone in it.

**Comment by mledoze** [comment]
@jhourlad please read this thread https://github.com/mledoze/countries/pull/106 to see why timezones were not added in this project.

**Comment by martin-ro** [comment]
Looking https://en.wikipedia.org/wiki/Continent there is no "Oceania" continent. The JSON label says region which makes sense to call the proximate islands of Australia Oceania though its not really a continent (depending who you ask).

Would it make sense to add a continent field according to one or more of the existing continent models?

cheers

**Comment by olimortimer** [comment]
As mentioned by others, it would be useful to be able to know if a country is within the EU.

https://en.wikipedia.org/wiki/Member_state_of_the_European_Union

**Comment by mledoze** [comment]
Hello @olimortimer, yes this has been asked before by others, but unfortunately, this kind of data does not suit this project. 

Please see this comment https://github.com/mledoze/countries/issues/6#issuecomment-71026239.

**Comment by maquejp** [comment]
Hello,
About adding the cities, you can maybe see what the Eurostat is doing : http://ec.europa.eu/eurostat/web/nuts/overview

**Comment by maquejp** [comment]
**Add a format: SQL (for main AND data)**
I have done it for the main (1st draft of the data attached to this comment)
I will work on the by country files to turn them into SQL too.

[mledoze.countries.sql.zip](https://github.com/mledoze/countries/files/325694/mledoze.countries.sql.zip)

**Comment by yackermann** [comment]
@maquejp Countries project ideology is KISS. Adding cities would majorly increase the size of the project, and would not serve according value. So I personally vote not on that idea.

About SQL DUMP: I don't see much of an issue to write a script that loads json into DB. There are far more then enough solutions online/stackoverflow. As addition it is a major security risk, to just load random sql dumps into databases. So I vote no on that one as well.

**Comment by cmacdonald** [comment]
What about the past? In using countries.json to analyse past pieces of text, I found "soviet" was not identified. Would adding a historical extension of the data with applicable dates be achievable in the long term?

**Comment by yackermann** [comment]
@cmacdonald Countries project goal is to provide basic information about current de-jure political state of the world. We provide basic, static information.

History will never be part of this project as it is contradicts the goal of this project to provide basic and static information, and due to single case use limitation.

**Comment by mledoze** [comment]
Hello @cmacdonald, I'm sorry but historical data are out of scope for this project.

**Comment by 8alery** [comment]
Hello! I think adding emoji for country flag will be nice. You can find emoji of flags here: http://emojiflags.com/

**Comment by dominicbartl** [comment]
If someone is searching for a country list with info if the country is in the EU: Here's a short script to append the info to the country list.

https://gist.github.com/Bartinger/f0507c786bad45cc942de471b1427e48

**Comment by mledoze** [comment]
@Bartinger thank you for that! :)

**Comment by pfwd** [comment]
Hi Thanks for a great project.
Are there any plans to add ISO 639-1 codes?  I'm looking for a way to adjust HTML language codes.
If the language section included the ISO 639-1 then it would be perfect for my use case.
I see that https://restcountries.eu has this support and its listed in the showcase.

**Comment by fayderflorez** [comment]
Hi @pfwd I started restcountries.eu based on @mledoze's great data some years ago but then I started to maintain my own version. That's why they are not much similar anymore.

**Comment by marc-ed-raffalli** [comment]
Hi @mledoze, how accurate should be the country name translations? Should it be 100% or it would be ok from an online translator.

**Comment by mledoze** [comment]
> Hi @mledoze, how accurate should be the country name translations? Should it be 100% or it would be ok from an online translator.

Hello @marc-ed-raffalli, I am sorry but I don't understand your question. The translations currently available in this project were gathered from various sources (mainly from Wikipedia). I hope that they are correct but I cannot guarantee that they are 100% correct.

**Comment by marc-ed-raffalli** [comment]
Hello @mledoze, I wanted to know if the translation has to be exact or is it ok to have translation from online translator like Google Translate (which may produce inaccurate output). Then, if it is ok, what would be the list of targeted languages?

**Comment by mledoze** [comment]
Well in that case, the translations have to be exact :)

There is no list of targeted languages, any language can be added to the list of translations.

**Comment by marc-ed-raffalli** [comment]
Good to know, I'll look for better translation sources :)

**Comment by marc-ed-raffalli** [comment]
Hi @mledoze 
I found translations for a good number country names and capitals e.g. London => Londres.
In https://github.com/mledoze/countries/issues/263#issuecomment-359933770 you said planning for 2.0. I would propose to change capital to a multi dimensional array too, will send a PR once ok.
```js
capital:{
  en: 'London',
  fr: 'Londres'
}
```

**Comment by mledoze** [comment]
> I would propose to change capital to a multi dimensional array

@marc-ed-raffalli this change was already made by @blumk in https://github.com/mledoze/countries/commit/d9e81cc083322b492c18b17faad54d5a3e4a7ff6. But translations for countries capital(s) are welcome :)

**Comment by marc-ed-raffalli** [comment]
@mledoze The change allows to support multiple capitals as an array of strings, does not allow a mapping to multiple languages. I would propose something like:
```
capital: {
  en: ['Pretoria', 'Cape Town', 'Bloemfontein'],
  fr: ['Pretoria', 'Le Cap', 'Bloemfontein'],
  ru: ['Претория', 'Кейптаун', 'Блумфонтейн'],
  zh: ['比勒陀利亚', '开普敦', '布隆方丹']
}
```
**edited** with better example

**Comment by blumk** [comment]
Looking at the current `countries.json` implementation I'd suggest refactoring translations into separate *.json files, e.g. `fra.json`:
```
{
  "cca3": "ZAF",
  "name": {"official": "R\u00e9publique d'Afrique du Sud", "common": "Afrique du Sud"},
  "capital": ['Pretoria', 'Le Cap', 'Bloemfontein'],
  "demonym": "Sud-Africain"
},
...
```
Creating separate language files brings the following benefits:
- Current implementation has a "translations" property. Adding capitals would lead to the following confusion: translation of what? (Name, capital?). Dedicated files solves this problem (see example above)
- Languages shouldn't pollute the main `countries.json` file since there are 6000+ languages. Adding a new language is difficult (see Arabic translations PR)
- You can give a single language file to a native speaker and she can confirm the correctness without knowing too much about the `countries.json` file
- Better support for lazy loading (e.g. two supported languages means loading two *.json files)

@marc-ed-raffalli @mledoze What do you guys think of this approach?

**Comment by marc-ed-raffalli** [comment]
@blumk 
> Better support for lazy loading (e.g. two supported languages means loading two *.json files)

Do you mean 3 files? one pure data (language agnostic), and two translation files

**Comment by blumk** [comment]
@marc-ed-raffalli 

> Do you mean 3 files? one pure data (language agnostic), and two translation files

Yes. Adding support for for two additional languages (`countries.json` defaults to `English`) means that you'll have to load `countries.json` and e.g. `fra.json` and `slk.json` in order to support `ENG`, `FRA`, `SLK`.

**Comment by marc-ed-raffalli** [comment]
@blumk 
I like the idea of separate language files. I would even push the idea and have all languages into dedicated files (including `ENG`) since not all apps necessarily support English :)

Let's see what's the feedback from the other contributors

**Comment by mledoze** [comment]
@blumk 
> Looking at the current countries.json implementation I'd suggest refactoring translations into separate *.json files

I find this idea interesting, but what would be your proposal for the directories/files structure? Given your example, would the `fra.json` file be in a `data/zaf` folder?

**Comment by blumk** [comment]
@mledoze 
> I find this idea interesting, but what would be your proposal for the directories/files structure? Given your example, would the fra.json file be in a data/zaf folder?

I guess there are two options:
- one translation file per language containing all 250 records: you can use a folder  like `data` | `translations` | `i18n` containing all translation files
- one translation file per language and record: in that case subfolders like `data/zaf` might be the way to go. However you end up having n*250 files

I might prefer one file per language containing all 250 records for the following reasons:
- it use case driven: e.g. drop down list of all countries in French, a map with all capitals in Dutch.
- one language, one location: you can give a single language file to a translator, you'll only need to load one file

E.g.
```
{
  "cca3": "ZAF",
  "name": {"official": "R\u00e9publique d'Afrique du Sud", "common": "Afrique du Sud"},
  "capital": ['Pretoria', 'Le Cap', 'Bloemfontein'],
  "demonym": "Sud-Africain"
},
{
  "cca3": "FRA",
  "name": ...
},
...
```

**Comment by ColinH** [comment]
My suggestion on "what to add next" would be NDD and IDD prefixes.

**Comment by mledoze** [comment]
> My suggestion on "what to add next" would be NDD and IDD prefixes.

Thank your for your suggestion, I'll look into it.

**Comment by mledoze** [comment]
@blumk I would like to move forward with your proposal to refactor translations to separate files; could you please open a new issue with your proposal? Thank you very much.

**Comment by lupinitylabs** [comment]
One thing I would really like to see added is the most commonly used native language / administrative language used in the country.

For example, Norway has three different, unweighted native names:
`"native":{"nno":{"official":"Kongeriket Noreg","common":"Noreg"},"nob":{"official":"Kongeriket Norge","common":"Norge"},"smi":{"official":"Norgga gonagasriika","common":"Norgga"}}`

And there is no way to tell which one of the three is the commonly used, for example in language or country choosers. In this example, Norge would obviously be the way to go, not Noreg or Norgga.

Can we either have a key for the official native name or a weighted/ranked list of languages in the country?

Also, there are altSpellings:
`"altSpellings":["NO","Norge","Noreg","Kingdom of Norway","Kongeriket Norge","Kongeriket Noreg"]`

But I don't feel comfortable trusting on the first items in the list to be the most common, also it's pretty much a mix of abbreviations and languages, which is generally not very helpful in this case. Also, Norgga from the native array above is missing in the altSpellings array, which is confusing?

**Comment by reknoZ** [comment]
Hello. Have you considered adding the date a country became a UN Member? Or if the country was an original UN Member?
